code/modules/projectiles/gun_helpers.dm

```diff
@@ -41,6 +41,8 @@
 
 //tactical reloads
 /obj/item/weapon/gun/afterattack(atom/target, mob/user, has_proximity, click_parameters)
+	if(!has_proximity)
+		return FALSE
```

Comment:
all the other returns are calling parent. Either this needs to call parent as well, or simply call parent at the start of the proc if there's no actual reason why its being called after.